featuretools_tsfresh_primitives/primitives/fft_aggregated.py

```diff
@@ -30,3 +35,115 @@ def function(x):
+        This limitation does not exist when using an offset alias to define `window_length`. In fact,
+        if the data has a uniform sampling frequency, it is preferable to use a numeric `gap` as it is more
+        efficient.
+
+    Docstring source:
+    https://tsfresh.readthedocs.io/en/latest/api/tsfresh.feature_extraction.html#tsfresh.feature_extraction.feature_calculators.fft_aggregated
+
+    Examples:
+        >>> import pandas as pd
+        >>> datetimes = pd.date_range(start="2019-01-01", freq="1D", periods=10)
+        >>> stffta = ShortTermFftAggregated(aggtype="centroid")
+        >>> output = stffta(datetimes, [1, 2, 4, 8, 4, 2, 1, 2, 4, 8]).tolist()
+        >>> [round(x, 5) for x in output]
+        [0.0, 0.25, 0.27429, 0.27429, 0.2, 0.27429, 0.27429, 0.16667, 0.27429, 0.27429]
+
+        We can also control the gap before the rolling calculation.
+
+        >>> stffta = ShortTermFftAggregated(aggtype="centroid", gap=1)
+        >>> output = stffta(datetimes, [1, 2, 4, 8, 4, 2, 1, 2, 4, 8]).tolist()
+        >>> [round(x, 5) for x in output]
+        [nan, nan, nan, 0.27429, 0.27429, 0.2, 0.27429, 0.27429, 0.16667, 0.27429]
+
+        We can also control the aggregation type.
+
+        >>> stffta = ShortTermFftAggregated(aggtype="variance")
+        >>> output = stffta(datetimes, [1, 2, 4, 8, 4, 2, 1, 2, 4, 8]).tolist()
+        >>> [round(x, 5) for x in output]
+        [0.0, 0.1875, 0.19906, 0.19906, 0.16, 0.19906, 0.19906, 0.13889, 0.19906, 0.19906]
+    """
+
+    name = "short_term_fft_aggregated"
+    input_types = [
+        ColumnSchema(logical_type=Datetime, semantic_tags={"time_index"}),
+        ColumnSchema(semantic_tags={"numeric"}),
+    ]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
+
+    def __init__(self, aggtype="centroid", window_length=3, gap=0, min_periods=0):
+        self.aggtype = aggtype
+        self.window_length = window_length
+        self.gap = gap
+        self.min_periods = min_periods
+
+    def get_function(self):
+        def short_term_fft_aggregated(datetime, numeric):
+            x = pd.Series(numeric.values, index=datetime.values)
+            rolled_series = roll_series_with_gap(
+                x, self.window_length, gap=self.gap, min_periods=self.min_periods
+            )
+            if isinstance(self.gap, str):
+                additional_args = (
+                    self.gap,
+                    self.appliable_fft_aggregated,
+                    self.min_periods,
+                )
+                return rolled_series.apply(
+                    apply_roll_with_offset_gap,
+                    args=additional_args,
+                ).values
+            return rolled_series.apply(self.appliable_fft_aggregated).values
+
+        return short_term_fft_aggregated
+
+    def appliable_fft_aggregated(self, series):
+        param = [{"aggtype": self.aggtype}]
+        return list(fft_aggregated(series.values, param=param))[0][1]
```

Comment:
Last question - the `fft_aggregated` documentation (also the `fft_coefficient docs) say that the function returns a `pd.Series`. I'm a little unclear on what is in the series, and why we have to convert to a list and then select list items like this.

Unfortunately I can't seem to get this repo to install on my M1 Mac, so I'm not able to load and test on my own to understand this...

Reply:
after doing some poking around it looks like it was written like this because it returns a zip object

Reply:
I see...